src/hash.cpp

```diff
@@ -29,18 +29,13 @@ extern "C" {
 
 #ifndef __APPLE__
 
-#include <openssl/sha.h>
-#include <openssl/md5.h>
-
-static_assert(utils::md5::DIGEST_SIZE == MD5_DIGEST_LENGTH, "Constants mismatch");
-static_assert(utils::sha1::DIGEST_SIZE == SHA_DIGEST_LENGTH, "Constants mismatch");
+#include <openssl/evp.h>
```

Comment:
Why did you remove the MD5 static_assert here but not in the Apple section? You should probably remove either both or neither.

Reply:
Because the new evp header doesn't have the constant to compare against anymore, and the evp header isn't used on Apple systems.

Reply:
I don't follow? Are there multiple types of MD5 hashes?